services/net/contentmigration/ContentMigrationAction.cs

```diff
@@ -130,19 +131,20 @@ public ContentMigrationAction(MigrationSourceContext sourceContext, ContentMigra
             {
                 var baseFilter = contentMigrator.GetBaseFilter();
                 IQueryable<NewsItem> items = GetFilteredNewsItems(_sourceContext.NewsItems, baseFilter, skip, count, manager.Ingest.LastRanOn, importDateStart, importDateEnd, creationDateOfLastImport);
-                var lastNewsItem = items.Last();
                 count = items.Count();
+                if (count == 0) break;
+
+                var lastNewsItem = items.Last();
```

Comment:
What is this used for?

Reply:
There are two loops happening here: 
First loop is picking up a manageable chunk of records each time. Second loop iterates over each item in the "chunk". `skip` is used to manage how many records we skip in the fetch from the database as we have (in theory) already migrated those. The `skip` value is used as the value for the `OFFSET` when it gets down to the `SQL` call.